src/backend/distributed/connection/connection_management.c

```diff
@@ -290,6 +290,15 @@ StartNodeUserDatabaseConnection(uint32 flags, const char *hostname, int32 port,
 		strlcpy(key.database, CurrentDatabaseName(), NAMEDATALEN);
 	}
 
+	if (flags & REQUIRE_REPLICATION_CONNECTION_PARAM)
+	{
+		key.replicationConnParam = true;
+	}
+	else
+	{
+		key.replicationConnParam = false;
+	}
+
```

Comment:
it would be good to ensure replication connections do not get cached, e.g. by returning true for replication connections in ShouldShutdownConnection

Reply:
@marcocitus : Added the `connection->requiresReplicationOption`  in ShouldShutdownConnection.